error/s2n_errno.c

```diff
@@ -248,6 +248,8 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_PQ_DISABLED, "Post-quantum crypto is disabled") \
     ERR_ENTRY(S2N_ERR_DUPLICATE_PSK_IDENTITIES, "The list of pre-shared keys provided contains duplicate psk identities") \
     ERR_ENTRY(S2N_ERR_OFFERED_PSKS_TOO_LONG, "The total pre-shared key data is too long to send over the wire") \
+    ERR_ENTRY(S2N_ERR_INVALID_PSK_ISSUE_TIME, "The pre-shared key issue time is in the future") \
```

Comment:
I wouldn't be so specific and keep the error code generic to be reusable, maybe just say: `Invalid pre-shared key issue time` and don't talk about the future aspect.

Reply:
We should consider making it generic enough to use to reject expired PSKs, since that's kind of the same problem.

Reply:
I created a generic error for these situations.